### PR TITLE
Memory dispatch

### PR DESCRIPTION
**Pull request by kunal-exuberant**

### CODE REVIEW

File: src/main/java/job/dispatcher/MemoryDispatcher.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+            JobContext currentJobExecutionContext = jobContextList.getFirst();
+            if(currentJobExecutionContext != null) {
+                if (currentJobExecutionContext.getNextScheduleEpoch() >= System.currentTimeMillis()) {
+                    delete(currentJobExecutionContext);
+                    log.info("Dispatching the job {}", currentJobExecutionContext.getJobId());
+                    httpRequest.post(currentJobExecutionContext.getCallbackUrl(), currentJobExecutionContext.getPayload());
+                    //updateDatastore(currentJobExecutionContext);
+                }
+                else {
+                    Thread.sleep(1);
```
Comment (bristy) [inline src/main/java/job/dispatcher/MemoryDispatcher.java:72]: Are we sleeping 1ms? Uninterruptibles.sleepUninterruptibly  can handle sleep better.

https://guava.dev/releases/19.0/api/docs/com/google/common/util/concurrent/Uninterruptibles.html#sleepUninterruptibly(long,%20java.util.concurrent.TimeUnit)

File: src/main/java/job/dispatcher/MemoryDispatcher.java [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+
+        if(jobContextList.isEmpty()){
+            jobContextList.addFirst(newJob);
+            return;
+        }
+
+        int index = 0;
+        JobContext start = jobContextList.get(index);
+
+        if(start.getNextScheduleEpoch() > newJob.getNextScheduleEpoch()){
```
Comment (bristy) [inline src/main/java/job/dispatcher/MemoryDispatcher.java:41]: Isn't it better to have a PriorityQueue where jobs are prioritised based on scheduled epochs?

File: pom.xml [hunk 3]
```diff
@@ -9,13 +9,22 @@
...
-      <module>DAL</module>
-      <module>core</module>
-      <module>JobRegistration</module>
-      <module>service</module>
+        <module>JobRegistration</module>
+        <module>service</module>
+        <module>models</module>
+        <module>DAL</module>
+        <module>core</module>
+        <module>JobRegistration</module>
```
Comment (ankitsingh-wtg) [inline pom.xml]: JobRegistration is mentioned twice in this pom.

Comment (kunal-exuberant) [reply pom.xml]: removed

**Pull request merged**